### DISCUSSION TITLE
Clearing added files

### DISCUSSION THREAD

**Discussion by CouinCouin** [wqweto/ZipArchive#29] [General]
Hi Vladimir :) 

First, I wish you a happy new year  :)

I'm about building a 2 GB limit workaround, by creating multiple zip files, based on filesize addition before compression.
I add in a listbox, each filenames and destination (in the zip archive) and I cumulate their size. 
If the next file to add makes the cumulation over than 2GB, I add a separator line ("============="), I reset the counter and continue to add the files.

Then I read the listbox items through a For loop. 
I ".AddFile" each files since the line is not a separator. 
When the line is a separator, I make the zip archive and I increase the part number (used in the zip filenames). 

Here is my code : 

```
With m_oZip
	For iL = 0 To lstbox4.ListCount - 1
		If lstbox4.List(iL) <> "=============" Then
			MyLineArr = Split(lstbox4.List(iL), "|")
			.AddFile MyLineArr(0), MyLineArr(1)
			PendingFiles = True
		Else
			PartNumber = PartNumber + 1
			If PartNumber > 1 Then
				UserFilenamePart = Mid(UserFilename, 1, Len(UserFilename) - 4) & " part" & PartNumber & _
					Mid(UserFilename, Len(UserFilename) - 3, 4)
			End If
			bResult = m_oZip.CompressArchive(UserFilenamePart, Level:=lLevel)
			PendingFiles = False
		End If
	Next iL

	If PendingFiles = True Then
		bResult = m_oZip.CompressArchive(UserFilenamePart, Level:=lLevel)
	End If
End With
```

I makes well for example, 2 parts : 
MyZipFile.zip
MyZipFile part2.zip

But in the MyZipFile part2.zip, I have the files that are already in MyZipFile.zip . I though (when I created the code with For loop) thah CompressArchive command was clearing the pending files once comrpession done, but seems not. 

Is there a way to manually clear the ".AddFile" list ? 
Or should I create a function where I set a new "m_oZip" New cZipArchive, make the comrpessed file and Set m_oZip = Nothing ? 

Thanks for ideas :) 
Couin

**Comment by wqweto** [comment]
Hi,

In commit 1dc100c66c08d2bf902d267b296154c40f25e69a I just added setters for `FileCount` and `LastError` properties so now you can try setting `.FileCount = 0` right after calling `CompressArchive` to "forget" about already processed files.

Another option to start with an empty archive is by creating a new instance of `cZipArchive` after calling `CompressArchive` which always exists as plan B.

cheers,
&lt;/wqw>
p.s. Don't forget to update your `cZipArchive` class with latest one.

**Comment by CouinCouin** [comment]
Hi :)

Just before you anwser, I had made a sub to create a new zip file with files to compress (I think this is equivalent to the "Plan B") , something like that :

```
Public Sub CreateZipFile(ByVal TargetFilename As String, ByVal AddedFiles As String)
	AddedFiles = Mid(AddedFiles, 2) 'because the AddedFiles start with a separator)
	m_bCancel = False

	Dim dblTimer        As Double
	Dim bResult         As Boolean
	Dim sLastError      As String
	Dim lLevel          As Long

	dblTimer = Timer
	Set m_oZip = New cZipArchive
	m_bCancel = False
	lLevel = 8

	Dim myline As String
	Dim MyLineArr() As String
	Dim AddedFilesArr() As String
	Dim iL As Long

	With m_oZip
		If InStr(AddedFiles, "*") > 0 Then 'Case of more than one file to archive
			AddedFilesArr() = Split(AddedFiles, "*")
			For iL = 0 To UBound(AddedFilesArr)
				MyLineArr = Split(AddedFilesArr(iL), "|")
				.AddFile MyLineArr(0), MyLineArr(1)
			Next iL
		Else
			MyLineArr = Split(AddedFiles, "|")
			.AddFile MyLineArr(0), MyLineArr(1)
		End If
		bResult = m_oZip.CompressArchive(TargetFilename, Level:=lLevel)
	End With

	Set m_oZip = Nothing
End Sub
```

And I call it with :
`Call CreateZipFile(UserFilenamePart, AddedFiles)`

But your commit is BETTER 👍 I removed CreateZipFile sub and calls for putting .FileCount = 0 and it looks to give the attempted result. Thanks !
